homeassistant/components/jewish_calendar/service.py

```diff
@@ -0,0 +1,463 @@
+                vol.Required("month"): vol.All(
+                    vol.In(MONTH_INPUT_MAP.keys()), MONTH_INPUT_MAP.get
+                ),
+                vol.Required("day"): vol.Coerce(int),
+                **OUTPUT_SCHEMA,
+            }
+        ),
+        supports_response=SupportsResponse.ONLY,
+    )
+    async_set_service_schema(
+        hass,
+        DOMAIN,
+        "get_hebrew_date",
+        {
+            "fields": {
+                "year": YEAR_FIELD,
+                "month": {
+                    "example": month_names[0],
+                    "required": True,
+                    "selector": {"select": {"options": month_names}},
+                },
+                "day": {
+                    "example": 1,
+                    "required": True,
+                    "selector": {"number": {"min": 1, "max": 30, "mode": "box"}},
+                },
+                **OUTPUT_FIELDS,
+            },
+        },
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        "get_holidays",
+        get_holidays,
+        schema=vol.Schema(
+            vol.Or(
+                {
+                    vol.Optional("year"): vol.Coerce(int),
+                    vol.Optional("holidays"): vol.All(
+                        cv.ensure_list,
+                        [vol.In(HOLIDAY_INPUT_MAP.keys())],
+                        [HOLIDAY_INPUT_MAP.get],
+                    ),
+                    **OUTPUT_SCHEMA,
+                },
+                {
+                    vol.Optional("year"): vol.Coerce(int),
+                    vol.Optional("types"): vol.All(
+                        cv.ensure_list,
+                        [cv.enum(htables.HolidayTypes)],
+                    ),
+                    **OUTPUT_SCHEMA,
+                },
+            )
+        ),
+        supports_response=SupportsResponse.ONLY,
+    )
+    async_set_service_schema(
+        hass,
+        DOMAIN,
+        "get_holidays",
+        {
+            "fields": {
+                "year": YEAR_FIELD,
+                "types": TYPES_FIELD,
+                "holidays": {
+                    "selector": {
+                        "select": {
+                            "multiple": True,
+                            "options": holiday_names,
+                        }
+                    }
+                },
+                **OUTPUT_FIELDS,
+            },
+        },
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        "get_next_holiday",
+        get_next_holiday,
+        schema=vol.Schema(
+            {
+                vol.Optional("date"): cv.datetime,
+                vol.Optional("types"): vol.All(
+                    cv.ensure_list,
+                    [cv.enum(htables.HolidayTypes)],
+                ),
+                **OUTPUT_SCHEMA,
+            }
+        ),
+        supports_response=SupportsResponse.ONLY,
+    )
+    async_set_service_schema(
+        hass,
+        DOMAIN,
```

Comment:
What does `async_set_service_schema` do?

Reply:
It registers a dynamically-generated `services.yaml` entry.

See https://github.com/home-assistant/core/pull/111986#discussion_r1615356659